### ISSUE TITLE
Loss of remote control with Chromecast shortly after starting playback

### ISSUE THREAD

**Issue by brianjmurrell** [jellyfin/jellyfin-web#3541]
**Describe The Bug**
Loss of remote control with Chromecast shortly after starting playback.

**Steps To Reproduce**
1. Select media to playback.
2. Select Chromecast destination.
3. Play media.
4. Immediately try to control media: stop, or move the playback slider, etc. and observe that works
5. Wait a few minutes.  Maybe 10 for good measure.
6. Try to control media: pause, stop, playback slider, etc.
7. Notice you no longer have any control over the playback.  You can't even disconnect from the Chromecast.
8. Further notice that the only way to play another video is to quit the app (i.e. kill it, swipe it away from the chooser, etc.), open the Google Home app and stop casting from there, then open the jellyfin app, re-navigate to where you want to play, re-choose Chromecast, play new video.  This last part is super annoying.

**Expected Behavior**
I should not lose remote control with Chromecast playback

**Logs**
Where would I find Chromecast playback control logs?

**System (please complete the following information):**
 - Platform: Linux server, Android client
 - Browser: Chrome
 - Jellyfin Version: 10.7.7, 10.8.0-beta1

**Comment by gausie** [comment]
I also get this on android

**Comment by brianjmurrell** [comment]
What I find interesting about it is that the remote control works for a short while after starting but then stops working at some point.  So the functionality is there, it just loses it's ability after a period of time.

**Comment by gausie** [comment]
Mine is sometimes like that but at the moment is immediately lost

**Comment by xanderpetit01** [comment]
I also have the same issue. But sometimes it works fine and when i start the next movie i lose control immediately

**Comment by xanderpetit01** [comment]
Still happens, i have Found that if you leave your phone unlocked and keep the jellyfin app open it doesn't happen. I have checked all powersaving settings and all are disabled.

**Comment by wjackter** [comment]
Still happening and very annoying.  I recently switched from Emby but this broken functionally makes me question my decision.

**Comment by thornbill** [comment]
Does this only happen when casting from the Android app? Or does it also happen when casting from Chrome?

**Comment by wjackter** [comment]
I only do it from my Android app, have not tried doing it from Chrome.

Thank you,

Warren

**Comment by terrycalderbank** [comment]
I definitely experience this issue as well. Android App, Google pixel 6 Pro, google cast stable. You usually lose the playback bar on the bottom of the app as well.

Because it happens so often, my wife and I have started referring to this as "losing it". I have found that sometimes if you swipe up close the app and open it again, you get it back and can control it again with the playback bar on the bottom appearing again. If I remember correctly it would do this with DLNA as well.

I'm somewhat software savvy and can't find anything wrong in the logs so feel free to recommend any debugging methods because I don't think I'll be sharing my Jellyfin instance with anyone without being able to vouch for android cast ability.

**Comment by tristanmccarthy** [comment]
Also encountering this issue with jellyfish v10.8.9 running in Linux.
Using Android 13 and v2.4.4 of the jellyfin android app.

**Comment by markymark86** [comment]
Got this problem too. Chrome V113 on Windows 11. If I hit the chromecast icon and select remote control, I can adjust the chromecast volume but get no indication of the playback progress and can't pause or otherwise control the cast.

**Comment by terrycalderbank** [comment]
Linking in a relevant Reddit discussion:

"For the disappearing now playing controls, I debugged it down to jellyfin-web.

In nowPlayingBar.js, the hideNowPlayingBar function is being called by some event in the webView version of the page, that doesn't happen in chrome desktop. I figured out that I need to now deploy the jellyfin-server in order to even deploy changes to jellyfin-web. Additionally, debugging webView is kinda painful, so I haven't started it yet.

As for chromecast losing connection to controls, my suspicion is that some kind of api changed, and the libraries or code haven't been updated properly, since there's been practically no releases to chromecast in 2 years. Unstable is updating libs, but code hasn't been looked at much, and no jellyfin developers use chromecast anymore, which is why unstable won't play anything."

https://www.reddit.com/r/jellyfin/comments/11zsx2a/chromecast_problems_cant_control_always_idle/jjhzzm2?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=share_button

**Comment by GottZ** [comment]
i've noticed reproducible behavior.

1. from a cold start of the current Google play release, select the stream target Chromecast, wait until jellyfin displays it's logo with wall art of your content.
2. start content
3. if no controls are displayed immediately, try to open the remote via the cast icon.
4. now you either get:
  - a empty screen with a single misaligned volume slider that might actually work but does not show initial volume.
  - proper working controls with either misaligned volume without initial state as one line up, or proper volume with alignment.

in case of issues at this point, kill the app while jelly is running on Chromecast, relaunch the app and repeat from step 3.
jelly might close when killing the app. in that case start over from step 1.

going this route works 100% of the time for me.
I hope it enlightens a contributor about anything relatable.

another thing that does not work via Chromecast:
if you pause content, wait a couple seconds or minutes and resume play, the stream will terminate after the buffer runs out, eventually even resetting the time tracking to 0:00:00 in the progress.
I could only get around this issue by never hitting play but the reverse 10 seconds button.

and lastly, jelly does not reply to HDMI-CEC commands on Chromecast like pretty much all other media center solutions in the market.
(no play / pause from the tv remote).. (in comparison, YouTube launches a almost feature complete YouTube TV web Client with full TV remote support)

**Comment by terrycalderbank** [comment]
This reflects my experience exactly. Additionally, sometimes if external subtitles are on and I skip forward or backwards, those subtitles are now frozen on the screen with the live subtitles displayed above them. The more times you do it, the more frozen subtitles begin to stack up.

**Comment by Etzix** [comment]
I have had the exact same issue as the 2 above for so long now. Im considering just dropping jellyfin because of it but I don't know where else I would go. But when this happens (and it happens a lot) the app is useless.

**Comment by brianjmurrell** [comment]
I gave up on Chromecasting entirely because of this issue.  I now buy Google TV devices which I can Chromecast to if I want to, but it supports running [jellyfin-androidtv](https://github.com/jellyfin/jellyfin-androidtv) and works much better.

It has it's own crashes but is easy enough to restart from the Google TV screen.

It also doesn't let you [delete episodes from it's interface](https://github.com/jellyfin/jellyfin-androidtv/discussions/2368) but that is supposed to be fixed in 0.16 which is not yet out.

**Comment by mp-v2** [comment]
I also experience these issues: Remote control not working, subtitles stacking up, jellyfin app losing control of chromecast. Is someone able to resolve this? I understand this is an open source project but these are multiple project-breaking bugs that have gone unfixed for a year and a half (since at least april 2022)

**Comment by terrycalderbank** [comment]
Issue still present. Have switched to the  android TV app and it works a lot better but casting from mobile still has the same bug.

**Comment by Kitof** [comment]
Yes, very annoying. Actually, this is the only remaining feature missing for me to revoke my Plex Subscription...

**Comment by joepie91** [comment]
Looking at the code, I suspect that [this](https://github.com/jellyfin/jellyfin-web/blob/f6d63c2b842ebda841a185b50cc048082281ee2a/src/plugins/chromecastPlayer/plugin.js#L36) might be where the problem is occurring, at least nominally. If an internal error is reported for a connection attempt more than once, it seems that the player is silently removed. Unfortunately the original casting error does not seem to be logged anywhere - the error message is swallowed and [only a generic error is logged](https://github.com/jellyfin/jellyfin-web/blob/f6d63c2b842ebda841a185b50cc048082281ee2a/src/plugins/chromecastPlayer/plugin.js#L266) (the original error is provided as the first argument of `onLaunchError` there).

This doesn't explain *why* errors are occurring, of course, and it's possible that different people are experiencing different errors leading up to this, but it would at least seem to explain why it results in the player suddenly disappearing.

**Comment by daniels0056** [comment]
I'm new to Jellyfin but have done some Android work before. I was about to take a look at this today and while I can consistently reproduce this issue in the Play Store published app, I cannot reproduce this issue in when building from the current code base in the [jellyfin-android](https://github.com/jellyfin/jellyfin-android) project, at least not in the debug version.

**Comment by Mountaineer1024** [comment]
I believe this is still a bug, and as of April 12, we still don't have a dev looking at it: https://github.com/jellyfin/jellyfin-chromecast/issues/364#issuecomment-2050017086

**Comment by Jcnewby** [comment]
Android jellyfin app, play/googlecast movie to selected chromecast, lose access to player in app, beyond play button.  
Frontscreen shows cutdown player: play/pause FF and FBack, no Stop.

**Comment by Mao899** [comment]
![Image](https://github.com/user-attachments/assets/1bcdc790-5156-47dd-a8e3-903a33c5ef76)

If this is what you're seeing, it's not just Android or Chromecast. This is the Jellyfin Windows App casting to another instance of the Windows App.

**Comment by GottZ** [comment]
if anyone familiar with the codebase reads this, even fingerpointing at rough estimations of where to look in code would help.
the complexity of this project is exceptional thanks to emby.

**Comment by Mao899** [comment]
I've been messing with ports for DLNA, hosts, fixed IP addresses, no luck. I do keep coming up with an error in my logs that I can't seem to pin down, seems related. Haven't been able to pin down where it's getting the localhost IP address from.

"[23:35:13] [INF] [11] Emby.Server.Implementations.HttpServer.WebSocketManager: WS 127.0.0.1 request
[23:35:46] [INF] [27] Emby.Server.Implementations.HttpServer.WebSocketManager: WS 127.0.0.1 closed"

Did also come across this:

https://github.com/jellyfin/jellyfin-mpv-shim/commit/7fab96351ab3ea0b5611bba4b2d73fc42bd2b163#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

**Comment by Mao899** [comment]
In my case, I seem to have narrowed it down to my use of Docker and Cosmos Server - after adding the Host IP address and port to "Overwrite Host Header" in the URL options in Cosmos, I can now at least maintain remote control, enough to make it useable. 
But, the remote device still disconnects its' session after being inactive for some time, and needs to be physically brought back online for other Jellyfin devices on the network to see it under the Play On option again. Starting playback or restarting the Jellyfin app so far seems to get it back online.

**Comment by brianjmurrell** [comment]
@Mao899 So I don't have any of this other over-complication of Jellyfin by using Docker and _Cosmos Server_ (whatever that is) so I don't think any of those other [over-]complications of a Jellyfin server are relevant to this issue.

I have a plain old simple Jellyfin server running in a Linux system without Docker or any other abstractions (not even a proxy, reverse or otherwise) and I have this problem.  It seems pretty clear that this problem is rooted right within the Jellyfin server and it's Chromecast handling and not due to to other layers/wrappers/etc.

But really, this issue doesn't even affect me any more as I threw money at the problem and replaced all of my dumb Chromecast devices (which required I cast from a phone/tablet/etc.) with the newer Chromecast-with-Google-TV devices and run the jellyfin-android client directly on those devices.

Frankly, the (minimal) cost of upgrading all of those devices was more than worth my sanity in trying to deal with this issue.  Given that, I don't blame the Jellyfin developers for not addressing this issue.

**Comment by daniels0056** [comment]
@brianjmurrell I have the issue with the Chromecast with Google TV cast, it's not the solution. It's not hardware related, it's Jellyfin.

**Comment by brianjmurrell** [comment]
@daniels0056 Yes, I would imagine casting Jellyfin from a device to a Chromcast-with-Google-TV would have the same issues.  But I don't cast JF to that device.  I put the jellyfin-android app on it and run it directly on the device.  No issues whatsoever.

**Comment by daniels0056** [comment]
@brianjmurrell then you're comment is totally irrelevant to this issue.

**Comment by brianjmurrell** [comment]
@daniels0056 No it's not.  Given that this issue is mine, I am the OP, my comment was intended to update my issue and provide a cheap and easy work-around to anyone struggling with this issue.  The frustration that this issue causes is not worth the low cost of a CCwGTV, IMO.

**Comment by tseho** [comment]
The issue is still happening.

**Comment by Skalmansbror** [comment]
Yep, Im having the same issue.
Im running the latest jellyfin server om a windows 11 machine. I am running https om 8920 with valid certificates.

I am using the web interface as the standalone "app" that appear when you choose to place the web page on the phones desktop. I feel this makes a better experience than the playstore app. Whenever the remote control stops working, i can do a quick refresh of the "web page", and when it reloads in a second or so, i am usually show as being connected to the chromecast (filled in chromecast icon). 

The issue exist in the web interface using Chrome, and the playstore app too. 

I am able to login, list my media, and playback works perfectly when casting.

Its just the remote control to the chromecast that glitches. I am usually able to press "pause" once and then "play" once, or "skip forward" twice, but after that no inputs will registrer.

BUT! In the android pull down meny i have the little notification of what is playing, along with the basic playback controls. If I use these buttons, i have total control! Play, pause, as much as I want to..! I can also see the remote control popup and disappear in the web interface "behind" the android drop down meny whenever I use the drop down meny controls. I also have control over play/pause using the screen gestures featured on my oneplus phone's blank screen while in standby. No limitations there. 

Also, refreshing the interface will usually "reconnect" the chromecast and allow me some control again, for a little while. 

I hope this can help anyone to narrow down the issue?
